File: nest-cloud/src/data/mappers/ProductSizeMapper.ts
import DomainMapper from "src/domain/mappers/DomainMapper";
import { ProductSize } from "src/domain/model/ProductSize";
import { ProductSizeEntity } from "../entity/ProductSizeEntity";

export class ProductSizeMapper implements DomainMapper<ProductSize, ProductSizeEntity>{
    toDomain(entity: ProductSizeEntity): ProductSize {
        return new ProductSize(
            entity.id,
            entity.product_id,
            entity.size,
        )
    }
    fromDomain(domain: ProductSize): ProductSizeEntity {
        return new ProductSizeEntity(
            domain.sizeId,
            domain.productId,
            domain.size,
            null, null
        )
    }

}

File: android-app/data/src/main/java/com/starlight/module/data/datasource/mappers/ProductSizeMapper.kt
package com.starlight.module.data.datasource.mappers

import com.starlight.module.data.datasource.entity.ProductSizeEntity
import com.starlight.module.domain.mappers.DomainMapper
import com.starlight.module.domain.model.ProductSize

class ProductSizeMapper : DomainMapper<ProductSizeEntity, ProductSize> {
    override fun toDomain(entity: ProductSizeEntity): ProductSize {
        return ProductSize(
            entity.sizeId,
            entity.size
        )
    }

    override fun fromDomain(model: ProductSize): ProductSizeEntity {
        return ProductSizeEntity(
            model.id,
            model.size
        )
    }
}